src/App.js

```diff
@@ -1,26 +1,71 @@
-import activities from './resources/schedule.json'
-import dayjs from "dayjs";
+import activities from "./resources/schedule.json";
+import CurrentWeek from "./components/CurrentWeek";
+import { formatDate } from "./helper functions/dateFormat";
+import ListedActivities from "./components/ListedActivities";
 
 function App() {
+  const today = formatDate(new Date());
+  var tomorrow = new Date();
+  tomorrow.setDate(tomorrow.getDate() + 1);
+  var remDays = 6 - tomorrow.getDay();
+  var currentWeekStart = new Date();
+  var currentWeekEnd = new Date();
+  var nextWeekStart = new Date();
+  var nextWeekEnd = new Date();
+  currentWeekStart.setDate(tomorrow.getDate() + 1);
+  currentWeekEnd.setDate(tomorrow.getDate() + remDays);
+  nextWeekStart.setDate(currentWeekEnd.getDate() + 1);
+  nextWeekEnd.setDate(nextWeekStart.getDate() + 6);
+
+  var sortedSchedule = {
+    past: [],
+    today: [],
+    tomorrow: [],
+    currentWeek: [],
+    nextWeek: [],
+    future: [],
+  };
+  var sortSchedule = (activities) => {
+    activities.map((activity) => {
+      var formatedDate = formatDate(new Date(activity.startTime));
+      switch (true) {
+        case formatedDate < today:
+          sortedSchedule.past.push(activity);
+          break;
+        case formatedDate === today:
+          sortedSchedule.today.push(activity);
+          break;
+        case formatedDate === formatDate(tomorrow):
+          sortedSchedule.tomorrow.push(activity);
+          break;
+        case formatedDate >= formatDate(currentWeekStart) &&
+          formatedDate <= formatDate(currentWeekEnd):
+          sortedSchedule.currentWeek.push(activity);
+          break;
+        case formatedDate >= formatDate(nextWeekStart) &&
+          formatedDate <= formatDate(nextWeekEnd):
+          sortedSchedule.nextWeek.push(activity);
+          break;
+        default:
+          sortedSchedule.future.push(activity);
+          break;
+      }
+      return false;
+    });
+   
+  };
+  sortSchedule(activities);
```

Comment:
This will run whenever the component renders. Imagine you have 1000 items, and this component happens to render 3 times (because some contexts were updating), then this function will run 3 times and filter the activities 3 times.

If the items were 10,000 you can imagine how slow this will become on the UI.

You should reference https://reactjs.org/docs/hooks-reference.html#usememo

Figure out how you can use that to solve your problem.